core/src/main/java/org/kiwix/kiwixmobile/core/search/viewmodel/effects/SearchIntentProcessing.kt

```diff
@@ -35,14 +38,23 @@ data class SearchIntentProcessing(
 ) : SideEffect<Unit> {
   @TargetApi(VERSION_CODES.M)
   override fun invokeWith(activity: AppCompatActivity) {
-    if (intent != null) {
-      if (intent.hasExtra(Intent.EXTRA_PROCESS_TEXT)) {
-        actions.offer(Filter(intent.getStringExtra(Intent.EXTRA_PROCESS_TEXT)))
+    intent?.let {
+      actions.offer(
+        ScreenOrigin(
+          if (it.getBooleanExtra(
```

Comment:
get the if condition on 1 line, I generally do 
```
myVariable=
  if(condition) value1
  else value2
```
if it doesn't fit on one line and if the if condition is too long
```
myVariable=
  if(condition)
     value1
  else
     value2
```